### PR TITLE
Add table aws_backup_legal_hold closes #1439

### LINKED ISSUE #1439: Add table aws_backup_legal_hold
**References**
API Reference: 
https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/backup#Client.ListLegalHolds
https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/backup#Client.GetLegalHold

### PR DESCRIPTION
**Pull request by ParthaI**
# Integration test logs
<details>
  <summary>Logs</summary>

```
N/A
```
</details>

# Example query results
<details>
  <summary>Results</summary>

```
> select
  legal_hold_id,
  arn,
  creation_date,
  cancellation_date
from
  aws_backup_legal_hold;
+--------------------------------------+---------------------------------------------------------------------------------------+---------------------------+-------------------+
| legal_hold_id                        | arn                                                                                   | creation_date             | cancellation_date |
+--------------------------------------+---------------------------------------------------------------------------------------+---------------------------+-------------------+
| 2c72c4e3-2936-4d53-8eb6-fdafce991da6 | arn:aws:backup:us-east-1:384702153875:legal-hold:2c72c4e3-2936-4d53-8eb6-fdafce991da6 | 2022-12-01T14:27:17+05:30 | <null>            |
+--------------------------------------+---------------------------------------------------------------------------------------+---------------------------+-------------------+

Time: 6.1s. Rows fetched: 1. Hydrate calls: 0.
> select
  legal_hold_id,
  arn,
  creation_date,
  creation_date,
  retain_record_until
from
  aws_backup_legal_hold
where
  creation_date >= current_date - interval '1' day
order by
  creation_date;
+--------------------------------------+---------------------------------------------------------------------------------------+---------------------------+---------------------------+------------------
| legal_hold_id                        | arn                                                                                   | creation_date             | creation_date             | retain_record_unt
+--------------------------------------+---------------------------------------------------------------------------------------+---------------------------+---------------------------+------------------
| 2c72c4e3-2936-4d53-8eb6-fdafce991da6 | arn:aws:backup:us-east-1:384702153875:legal-hold:2c72c4e3-2936-4d53-8eb6-fdafce991da6 | 2022-12-01T14:27:17+05:30 | 2022-12-01T14:27:17+05:30 | <null>           
+--------------------------------------+---------------------------------------------------------------------------------------+---------------------------+---------------------------+------------------

Time: 2.4s. Rows fetched: 1. Hydrate calls: 1.
> select
  title,
  legal_hold_id,
  recovery_point_selection -> 'DateRange' ->> 'ToDate' as to_date,
  recovery_point_selection -> 'DateRange' ->> 'FromDate' as from_date,
  recovery_point_selection -> 'VaultNames' as vault_names,
  recovery_point_selection ->> 'ResourceIdentifiers' as resource_identifiers
from
  aws_backup_legal_hold;
+--------+--------------------------------------+--------------------------+----------------------+-------------+--------------------------------+
| title  | legal_hold_id                        | to_date                  | from_date            | vault_names | resource_identifiers           |
+--------+--------------------------------------+--------------------------+----------------------+-------------+--------------------------------+
| test43 | 2c72c4e3-2936-4d53-8eb6-fdafce991da6 | 2022-12-01T08:57:16.295Z | 2022-11-29T18:30:00Z | []          | ["arn:aws:ec2:*:*:instance/*"] |
+--------+--------------------------------------+--------------------------+----------------------+-------------+--------------------------------+

Time: 2.5s. Rows fetched: 1. Hydrate calls: 1.
> 
```
</details>

### CODE REVIEW

File: aws/table_aws_backup_legal_hold.go [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+	"github.com/turbot/steampipe-plugin-sdk/v4/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/v4/plugin/transform"
+)
+
+//// TABLE DEFINITION
+
+func tableAwsBackupLegalHold(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "aws_backup_legal_hold",
+		Description: "AWS Backup LegalHold",
```
Comment (cbruno10) [inline aws/table_aws_backup_legal_hold.go:20]:
```suggestion
		Description: "AWS Backup Legal Hold",
```

File: aws/table_aws_backup_legal_hold.go [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+				Description: "This specifies criteria to assign a set of resources, such as resource types or backup vaults.",
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getAwsBackupLegalHold,
+			},
+
+			// Steampipe standard columns
+			{
+				Name:        "title",
+				Description: resourceInterfaceDescription("title"),
+				Type:        proto.ColumnType_STRING,
```
Comment (cbruno10) [inline aws/table_aws_backup_legal_hold.go:81]: What resource property is used for `title`? Is there a column called `title` we get in the data?

File: aws/table_aws_backup_legal_hold.go [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+			},
+			{
+				Name:        "status",
+				Description: "This is the status of the legal hold. Statuses can be ACTIVE, CREATING, CANCELED, and CANCELING.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "cancellation_date",
+				Description: "This is the time in number format when legal hold was cancelled.",
+				Type:        proto.ColumnType_TIMESTAMP,
```
Comment (cbruno10) [inline aws/table_aws_backup_legal_hold.go:57]: What does the data look like for this column? Does it come as epoch time, or something else? If so, we should convert to the standard date/time format we use for other columns.

File: aws/table_aws_backup_legal_hold.go [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+			},
+			{
+				Name:        "description",
+				Description: "This is the description of a legal hold.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "retain_record_until",
+				Description: "This is the date and time until which the legal hold record will be retained.",
+				Type:        proto.ColumnType_STRING,
```
Comment (cbruno10) [inline aws/table_aws_backup_legal_hold.go:67]: What are some possible values for this column? Should it be a timestamp column, or is string appropriate?

File: docs/tables/aws_backup_legal_hold.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+  retain_record_until
+from
+  aws_backup_legal_hold
+where
+  creation_date <= current_date - interval '10' day
+order by
+  creation_date;
+```
+
+### Get recovery point selection details for legal holds
```
Comment (cbruno10) [inline docs/tables/aws_backup_legal_hold.md:36]:
```suggestion
### Get recovery point selection details for each legal hold
```

**Review by cbruno10 — CHANGES_REQUESTED** [review]
@ParthaI Please see review comments, thanks!

File: aws/table_aws_backup_legal_hold.go (hunk 2 above)
Comment (ParthaI) [reply aws/table_aws_backup_legal_hold.go:81]: We are getting property called `Title` from API itsetlf

File: aws/table_aws_backup_legal_hold.go (hunk 3 above)
Comment (ParthaI) [reply aws/table_aws_backup_legal_hold.go:57]: In Go SDK, we are getting the value of `CancellationData` of type `*time.Time`, it is not coming as an epoch time

File: aws/table_aws_backup_legal_hold.go (hunk 4 above)
Comment (ParthaI) [reply aws/table_aws_backup_legal_hold.go:67]: It will be a timestamp column

**Review by misraved — APPROVED** [review]

**Review by misraved — APPROVED** [review]

**Pull request merged**